src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedSimpleProgramEntryPointSymbol.cs

```diff
@@ -18,47 +19,19 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     internal sealed class SynthesizedSimpleProgramEntryPointSymbol : SourceMemberMethodSymbol
     {
         /// <summary>
-        /// A synthetic syntax tree associated with this method. An empty <see cref="CompilationUnitSyntax"/>
-        /// is the root of this tree. In several places in the compiler (like for example in 
-        /// <see cref="MethodCompiler.BindMethodBody(MethodSymbol, TypeCompilationState, DiagnosticBag, out ImportChain, out bool, out MethodBodySemanticModel.InitialState)"/>)
-        /// we expect that a <see cref="SourceMemberMethodSymbol"/> that contains user code is associated with 
-        /// a single syntax node. We are using the empty <see cref="CompilationUnitSyntax"/> for this purpose.
-        /// Also, since multiple compilation units contribute locals and local functions to a single scope, we
-        /// need a single syntax node that can be used as a <see cref="Binder.ScopeDesignator"/> for it without
-        /// creating an ambiguity.
-        /// Another dependency on a presence of a single syntax node is a <see cref="ExecutableCodeBinder"/> that is
-        /// now also responsible for building binders for the compound simple program body.
-        /// 
-        /// There is one-to-one correspondence between an instance of <see cref="SynthesizedSimpleProgramEntryPointSymbol"/> 
-        /// and the tree. Even though the <see cref="CompilationUnitSyntax"/> node it empty, logically it represents a
-        /// compound simple program body, the logical children are all real <see cref="CompilationUnitSyntax"/> nodes with
-        /// top-level statements in the compilation.
+        /// The corresponding <see cref="SingleTypeDeclaration"/>. 
         /// </summary>
-        private readonly CSharpSyntaxTree _syntheticSyntaxTree = new CSharpSyntaxTree.DummySyntaxTree();
-
-        /// <summary>
-        /// The first syntax tree with a top-level statement that is not a <see cref="LocalFunctionStatementSyntax"/>.
-        /// That is the tree that comes first in the logical method body. It is possible to have no tree like that. 
-        /// </summary>
-        internal readonly SyntaxTree? PrimarySyntaxTree;
+        SingleTypeDeclaration _declaration;
```

Comment:
>SingleTypeDeclaration [](start = 8, length = 21)

`private`

Reply:
```private readonly```

I will make this change in the next PR

---
In reply to: [397388541](https://github.com/dotnet/roslyn/pull/42648#discussion_r397388541) [](ancestors = 397388541)

Reply:
Addressed in [https://github.com/dotnet/roslyn/pull/42751](https://github.com/dotnet/roslyn/pull/42751)

---
In reply to: [397474586](https://github.com/dotnet/roslyn/pull/42648#discussion_r397474586) [](ancestors = 397474586,397388541)